st2common/st2common/services/workflows.py

```diff
@@ -535,17 +535,24 @@ def request_action_execution(wf_ex_db, task_ex_db, st2_ctx, ac_ex_req, delay=Non
     # Identify the runner for the action.
     runner_type_db = action_utils.get_runnertype_by_name(action_db.runner_type['name'])
 
+    # Identify action pack name
+    action_ref = task_ex_db.task_spec.get('spec').get('action')
```

Comment:
There's already an action_ref at line 519.  They're not the same?

Reply:
You did not address this yet.

Reply:
It is not the same. line 519 is get block of action data from action_execution_d_b. This is get action_ref = pack_name:action_name

Not sure about `You did not address this yet.` means. Changed `if else` block to one line.

Reply:
The line numbers may have shifted.  But if you go down two line from `def request_action_execution(wf_ex_db, task_ex_db, st2_ctx, ac_ex_req, delay=None)`, there's already an action_ref assigned.  Is it different than the action_ref begin assigned in this line.

Reply:
And if they are not the same, then you just overwrote action_ref that is saved to LiveActionDB below.

Reply:
Yes, it is different. Changed my variable name.

Reply:
I just manually tested this and they are the same value.  Can you show me an example where they are different.

Reply:
My bad.  I thought it is reading ActionExecution.action. Put log message and tested.  Committed change.

Reply:
Ok. Good you finally see it. I thought maybe I'm not clear enough.